File: answers/leetcode/Add Binary.py
class Solution(object):
    def addBinary(self, a, b):
        """
        :type a: str
        :type b: str
        :rtype: str
        """

        sum = [0 for i in range(max(len(a), len(b)))]
        for i in range(1, max(len(a), len(b)) + 1):
            if i <= len(a):
                sum[- i] += int(a[- i])
            if i <= len(b):
                sum[- i] += int(b[- i])
        carry = 0
        for i in reversed(range(len(sum))):
            carry += sum[i]
            sum[i] = carry % 2
            carry /= 2
        while carry:
            sum.insert(0, carry % 2)
            carry /= 2
        return "".join([str(i) for i in sum])

File: answers/leetcode/Add Binary.cpp
// 2014-04-20 AC
//@result 294 / 294 test cases passed. Status: Accepted Runtime: 4 ms Submitted: 0 minutes ago You are here! Your runtime beats 20.86% of cpp submissions.

class Solution {
public:
    string addBinary(string a, string b) {
		string ans;
		int carry = 0;
		size_t i;
		for (i = 0; i < a.length() && i < b.length(); ++ i)
		{
			int sum = (a[a.length() - 1 - i] - '0') + (b[b.length() - 1 - i] - '0') + carry;
			carry = sum / 2;
			sum %= 2;
			ans += static_cast<char>(sum + '0');
		}
		for (; i < a.length(); ++ i)
		{
			int sum = (a[a.length() - 1 - i] - '0') + carry;
			carry = sum / 2;
			sum %= 2;
			ans += static_cast<char>(sum + '0');
		}
		for (; i < b.length(); ++ i)
		{
			int sum = (b[b.length() - 1 - i] - '0') + carry;
			carry = sum / 2;
			sum %= 2;
			ans += static_cast<char>(sum + '0');
		}
		while (carry)
		{
			int sum = carry;
			carry = sum / 2;
			sum %= 2;
			ans += static_cast<char>(sum + '0');
		}
		std::reverse(ans.begin(), ans.end());
		return ans;
    }
};
